dist-build/wasm32-wasi.sh

```diff
@@ -1,37 +1,18 @@
 #! /bin/sh
 
-if [ -z "$WASI_LIBC" ]; then
-  for path in /opt/wasi-libc /opt/wasi-sysroot; do
-    if [ -d "$path" ]; then
-      export WASI_LIBC="$path"
-      break
-    fi
-  done
-fi
-if [ -z "$WASI_LIBC" ]; then
-  echo "Set WASI_LIBC to the path to the WASI libc sysroot" >&2
+if ! command -v wasienv >/dev/null; then
+  echo "Wasienv needs to be installed - https://github.com/wasienv/wasienv" >&2
   exit 1
 fi
 
-export PATH="/usr/local/opt/llvm/bin:$PATH"
-
 export PREFIX="$(pwd)/libsodium-wasm32-wasi"
 
 mkdir -p $PREFIX || exit 1
 
-export CC="clang"
-export CFLAGS="-DED25519_NONDETERMINISTIC=1 --target=wasm32-wasi --sysroot=${WASI_LIBC} -O2"
-export LDFLAGS="-s -Wl,--no-threads"
-export NM="llvm-nm"
-export AR="llvm-ar"
-export RANLIB="llvm-ranlib"
-export STRIP="llvm-strip"
+export CFLAGS="-DED25519_NONDETERMINISTIC=1 -O2"
 
 make distclean > /dev/null
 
-grep -q -F -- 'wasi' build-aux/config.sub || \
```

Comment:
This is necessary for Frankeinstein Linux systems, where people mix a base system containing old tools (here: old autotools) with backported or hand-compiled recent tools (here: LLVM). And unfortunately, this is a very common setup.